### DISCUSSION TITLE
Is it possible to mute all notes at the interval of multiple songs in fluidsynth binary?

### DISCUSSION THREAD

**Discussion by ghost** [FluidSynth/fluidsynth#1133] [Ideas]
Dear Tom M,
I've found that the fluidsynth binary does not mute all notes automatically when the song changes. For example, I play `A.mid` and `B.mid` in order. The final note of `A.mid` will not be muted and it will sustain for a few seconds when `B.mid` starts. The sound is very weird.
Is it possible to improve this?

**Comment by derselbst** [comment]
It would be helpful if you could provide demo MIDIs, a Soundfont and let us know, which version of fluidsynth you're using.

**Reply by ghost** [reply]
SoundFont: FluidR3_GM
MIDI: `C:\windows\media\onestop.mid` played twice
Version: 2.2.8
Here is my audio:
[onestop-midi.ogg.zip](https://github.com/FluidSynth/fluidsynth/files/9250315/onestop.zip)
There is something wrong at 4'07" . The MIDI program of the final notes is `FX 5 (Brightness)`. Its sound gained for a second when the 2nd time of `onestop.mid` began.

**Reply by derselbst** [reply]
Ok, I can reproduce it and I agree it doesn't sound right. I tried other MIDIs and they don't show that behavior. Sounds as if an event is still pending and getting a volume change or so. I need to investigate.

**Reply by derselbst** [reply]
I found some time looking into this: As you correctly said, the second MIDI channel is playing program 100 (Brightness) at the very end. The MIDI uses CC7 (main volume) to create a fadeout effect. When the last note of that channel receives its noteoff, the voices still keep playing in release phase for about 5 seconds. The player currently doesn't wait for all voices to end before starting the next song (see #24). So, when starting the next song, the main volume is set back to 125 (i.e. almost full open). Since the voices are still playing, we hear how the player resets the main volume of all channels, therefore we hear the pending voices.

One could address this by continue rendering audio after the player has finished until `fluid_synth_get_active_voice_count()` return 0. This will however not take care of any pending reverb.
